front-end/src/Views/ProfilePage.js

```diff
@@ -608,4 +638,274 @@ export default class ProfilePage extends Component {
     else
       return false;
   }
+
+  handleCheckCurrentlyNotFollow = (id) => {
+
+    var check = 0;
+    
+    axios.get(`${config.API_URL}${config.User_Path}${id}/`, { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
+        .then(res => {
+
+          var nameName = res.data.profile[0].name + " " +res.data.profile[0].middle_name + " " + res.data.profile[0].last_name;
+
+          this.state.followers.forEach(item => {
+            const member = item;
+            if(member.name + " " + member.last_name === nameName){
+              check = 1;
+              
+            }
+            
+          });
+        });
+
+        console.log("1")
+        if(check === 1){
+          console.log("2")
+          return false
+        }
+        else{
+          console.log("3")
+          return true
+        }
+  } 
+
+  renderFollow() {
+

```

Comment:
You shouldn't make api calls every time something renders, it's better to put it in component did mount